src/app/CommandRefLookupTable.h

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/ConcreteCommandPath.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/Optional.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * @class CommandRefLookupTable
+ *
+ * @brief Allows looking up CommandRef using the requested ConcreteCommandPath.
+ *
+ * While there are faster implementations, right now batch commands are capped at a low number due to
+ * message size constrains. All commands need to be contained within a single InvokeRequest. In
+ * practice this is less than 60 commands.
+ *
+ */
+class CommandRefLookupTable
+{
+public:
+    struct CommandRefTableEntry
+    {
+        ConcreteCommandPath requestPath = ConcreteCommandPath(0, 0, 0);
+        Optional<uint16_t> ref;
```

Comment:
With moving to zero-as-default, then this would be non-optional, and would save memory

Reply:
This can be done at a future time, when the spec is changed to reflect this